pytest_jira.py

```diff
@@ -230,19 +230,35 @@ def get_url(self):
 class JiraMarkerReporter(object):
```

Comment:
General flow here, how the plugin works here, is:
  Per each test-case do:
    * collect all issue IDs associated with test-case
    * fetch issues from server considering miss-strategy when issue was not found
    * make a decision on test-case if skip or not

This class is supposed to collect and provide list of IDs which are related / applied to single test-case.
So in order to preserve work-flow I mentioned above, I would prefer to do not add `connection` parameter, neither fetching issues from server and neither making decisions about  skipping.

I would rather create some structure, or class to hold the ID, predicate and latter on data fetched from server.
Then you can move skipping logic at place where decision is being done.

Reply:
Hmm. Would best way then to change it to collect array of issue-tuples instead array of issues as test might have multiple jira markers with different condition? This would be backwards incompatible though. 

```python
@pytest.mark.jira("ABC-1000", "ABC-1001", skipif=<False or lambda returning False>)
@pytest.mark.jira("ABC-1002", "ABC-1003", skipif=<True or lambda returning True>)
def test_fail():
  assert False
```

JiraMarkerReporter.get_jira_issues() would now return 
```python
[
  ("ABC-1000",  skipif=<False or lambda returning False>), 
  ("ABC-1001", skipif=<False or lambda returning False>),
  ("ABC-1002", skipif=<True or lambda returning True>),
  ("ABC-1003", skipif=<True or lambda returning True>)
]
```

...And for a moment I thought this was just a small harmless little pull request :D